### PR TITLE
Implement unidirectional sequence GRU kernel.

### PR DESCRIPTION
**Pull request by sixuewang**
Implement unidirectional sequence GRU kernel supporting float eval, and add basic unit tests.

### CODE REVIEW

**Review by renjie-liu — DISMISSED** [review]
lgtm, but wait for Jian's review as well

**Comment by rthadur** [comment]
> lgtm, but wait for Jian's review as well

@jianlijianli gentle ping

File: tensorflow/lite/experimental/kernels/unidirectional_sequence_gru_test.cc [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  const int n_input = 2;
+  const int n_output = 3;
+
+  GRUOpModel m({{n_time, n_batch, n_input},
+                {n_batch, n_output},
+                {2 * n_output, n_input + n_output},
+                {2 * n_output},
+                {n_output, n_input + n_output},
+                {n_output}});
+  m.SetInput({0.89495724, 0.34482682, 0.68505806, 0.7135783, 0.3167085,
```
Comment (jianlijianli) [inline tensorflow/lite/experimental/kernels/unidirectional_sequence_gru_test.cc]: Are those random numbers or from some know public model? If they are random numbers, please just add a simple comment. Thanks.

File: tensorflow/lite/experimental/kernels/gru_cell.cc [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  concat_arrays_shapes.push_back(&input_shape);
+  concat_arrays_shapes.push_back(&state_shape);
+  tflite::ConcatenationParams concat_params;
+  concat_params.axis = 1;
+  concat_params.inputs_count = concat_arrays_data.size();
+  Concatenation(concat_params, &(concat_arrays_shapes[0]),
+                &(concat_arrays_data[0]), concat_shape, concat);
+
+  // [r u] = [x h] * gate_weight + gate_bias
+  tflite::FullyConnectedParams fc_params;
```
Comment (jianlijianli) [inline tensorflow/lite/experimental/kernels/gru_cell.cc]: Can we move this out of the kernel and put it in Eval()?

**Review by jianlijianli — DISMISSED** [review]
Apologies for the delay and thanks for adding GRU. The logic looks good and thanks Renjie for reviewing. Just a few small comments.

File: tensorflow/lite/experimental/kernels/unidirectional_sequence_gru_test.cc (hunk 1 above)
Comment (sixuewang) [reply tensorflow/lite/experimental/kernels/unidirectional_sequence_gru_test.cc]: Done

Comment (sixuewang) [reply tensorflow/lite/experimental/kernels/unidirectional_sequence_gru_test.cc]: Done

File: tensorflow/lite/experimental/kernels/gru_cell.cc (hunk 2 above)
Comment (sixuewang) [reply tensorflow/lite/experimental/kernels/gru_cell.cc]: Done

**Review by jianlijianli — DISMISSED** [review]
Thanks!

File: tensorflow/lite/experimental/kernels/gru_cell.cc [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  concat_arrays_shapes.push_back(&input_shape);
+  concat_arrays_shapes.push_back(&state_shape);
+  tflite::ConcatenationParams concat_params;
+  concat_params.axis = 1;
+  concat_params.inputs_count = concat_arrays_data.size();
+  Concatenation(concat_params, &(concat_arrays_shapes[0]),
+                &(concat_arrays_data[0]), concat_shape, concat);
+
+  // [r u] = [x h] * gate_weight + gate_bias
+  FullyConnected(fc_params, concat_shape, concat, gate_weight_shape,
```
Comment (renjie-liu) [inline tensorflow/lite/experimental/kernels/gru_cell.cc]: I think you need to sync to the head to match with the new signature

**Review by renjie-liu — DISMISSED** [review]

File: tensorflow/lite/experimental/kernels/gru_cell.cc (hunk 3 above)
Comment (sixuewang) [reply tensorflow/lite/experimental/kernels/gru_cell.cc]: Done

**Review by renjie-liu — APPROVED** [review]

**Review by jianlijianli — APPROVED** [review]

**Pull request merged**